sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -582,7 +586,7 @@ object IntervalUtils {
             case _ if '0' <= b && b <= '9' && fractionScale > 0 =>
               fraction += (b - '0') * fractionScale
               fractionScale /= 10
-            case ' ' =>
+            case ' ' if !pointPrefixed || fractionScale < initialFractionScale =>
```

Comment:
since `1.` is allowed, I don't think `fractionScale < initialFractionScale` is a valid check any more.

Reply:
we need this, the cases are (`1.0`, `0.1`, `.1`,  `1.`) + (`.`), this condition is equal to `!(pointPrefixed && fractionScale == initialFractionScale)`

Reply:
rm `fractionScale < initialFractionScale` forbiding `.1` case

Reply:
ah i see!